src/main/java/org/shanerx/tradeshop/util/JsonConfiguration.java

```diff
@@ -56,55 +60,40 @@ public JsonConfiguration(Chunk c) {
 				throw new RuntimeException(exception);
 			}
 		}
-	}
 
-	private void writeJSON(String path, String value) {
-		JsonObject obj = new JsonObject();
-		obj.add(path, new JsonParser().parse(value));
 		try {
-			FileWriter fileWriter = new FileWriter(this.file);
-			fileWriter.write(obj.getAsString());
-			fileWriter.flush();
-			fileWriter.close();
-		} catch (Exception e) {
-			throw new RuntimeException(e);
-		}
-	}
-
-	private JsonObject readJSON(String path) {
-		try {
-			return new JsonParser().parse(new FileReader(file)).getAsJsonObject();
+			jsonObj = new JsonParser().parse(new FileReader(file)).getAsJsonObject();
 		} catch (FileNotFoundException e) {
 			throw new RuntimeException(e);
 		}
 	}
 
-	// CAREFUL: DOES NOT OUTPUT JSON! USE ONLY FOR FILE NAMES!
-	private String serializeChunk(Chunk chunk) {
-		String world = chunk.getWorld().getName();
-		int x = chunk.getX(), z = chunk.getZ();
+	private void setShop(Shop shop) {
+		Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
+		String shopJson = gson.toJson(shop), toSaveStr = "";
 
-		return "c" + div + world + div + x + div + z;
-	}
+		jsonObj.add(shop.getShopLocationAsSL().serialize(), gson.toJsonTree(shop));
```

Comment:
uh... what does SL mean?

Reply:
@SparklingComet ShopLocation didn't want a 2 mile long method also wierd to read shop location as shop location

Reply:
Call it #getShopLocation ?

Reply:
#getShopLocation returns a Location object because were going to serialize Shop with gson and id rather that get serialized so that ShopLocation doesnt need to be tweaked to work well with the gson serialization.